### DISCUSSION TITLE
How to make icons that will fit with Fallout theme

### DISCUSSION THREAD

**Discussion by thunkadelic** [shvchk/fallout-grub-theme#27] [Q&A] [answered]
I love your Fallout theme! I've been adding some distributions that don't already have Fallout icons. Took a while to figure out how and where to put the icons and class distros to use them, both in Grub2 and with the BLS (boot loader specification) in Fedora. Got that working! They don't quite fit your Fallout style though.

Here's my thought: I'd like to add icons for some distros that don't already have them. I can probably manipulate what I've got to approximate your Fallout style, but I thought that if you already have some sort of filter that you apply to existing icons, I could use that to produce some that really fit in.

Longer term idea: if it's some kind of automatable filter, I'd like to set something up so that the Fallout theme, and others themes, if their creators so desired, could use such a filter to grab icons and produce matching ones automatically. Still working on that thought. In any case, starting with how you produce your icons might be a good first step. 

Thanks for your thoughts, and for an awesome theme!

(picture of my grub/BLS menu with my handmade icons attached)
![grub-BLS menu with added icons](https://user-images.githubusercontent.com/69141428/119790410-c3847680-be88-11eb-998c-d615af5bcd6c.jpg)

**Comment by thunkadelic** [comment]
Making progress on the conformity to the 
Fallout icon style, and on the automation of creating the icons. 
Can't seem to get that shade of green right, though.

GraphicsMagick command for turning ordinary, everyday icons (in this case the KDE neon icon) into a Fallout theme icons:

(Edit: Oops. Didn't paste that command in, did I? Here it is now.)

gm convert neon-larger.png -resize 32x32 -bordercolor transparent -border 10x10 -colors 2 -fuzz 50% -fill '#63DA72ff' -opaque blue 1neon.png

![MakingProgress](https://user-images.githubusercontent.com/69141428/120104977-dc508e80-c10b-11eb-858a-c91b37963ff9.jpg)

**Comment by Teknoist** [comment]
command not for me can you help me ?

**Comment by Teknoist** [comment]
![kaspersky2](https://user-images.githubusercontent.com/37031361/203631323-74c28d18-c634-4ffb-b79a-47c7ec96a048.PNG)

**Comment by Teknoist** [comment]
![kaspersky](https://user-images.githubusercontent.com/37031361/203631356-ce6ab895-55c0-40ba-b901-71e227964105.png)

**Comment by thunkadelic** [comment]
Hello Teknoist, my  command isn't working for you? Well, if that's the icon that you want converted, I'll work on it. (Sometime soon, but not at this moment.) 

For making my command easier to use, can you tell me what exact command you used and what output you got?

**Comment by Teknoist** [comment]
ı use 
`gm convert neon-larger.png -resize 32x32 -bordercolor transparent -border 10x10 -colors 2 -fuzz 50% -fill '#63DA72ff' -opaque blue 1neon.png`
code and give me that first photo ı uploaded

**Comment by shvchk** [comment, marked as answer]
Hey guys :wink: 
I did icons conversion some years ago, so I don't remember exact command, but I would try one of these:
- GIMP → Color → Colorize → Color → HTML notation → `67d97a`  
  For batch processing with GIMP use [BIMP](https://github.com/alessandrofrancesconi/gimp-plugin-bimp)\* `gimp-drawable-colorize-hsl` with hue: 130, saturation 60, lightness 25.5. These HSL values are from GIMP Colorize dialog for `#67d97a`, with hue value converted to degrees by multiplying it by 360.
- `coloration -h 130 -s 60 -l 25.5 in.png out.png` from [Fred's ImageMagick Scripts](http://www.fmwconcepts.com/imagemagick/coloration)
- `convert in.png -colorspace RGB -colorspace Gray -fill '#67d97a' -tint 100 out.png`
- `convert in.png -colorspace RGB -colorspace Gray +level-colors black,'#67d97a' out.png`  
  You might also want to increase brightness with [`-brightness-contrast x`](https://imagemagick.org/script/command-line-options.php#brightness-contrast), with `x` in range from -100 to +100, 0 leaves brightness as is, and +100 turns any color white (max brighten):  
  `convert in.png -colorspace RGB -colorspace Gray -brightness-contrast +100 +level-colors black,'#67d97a' out.png`

Use `for` loop for batch processing with CLI tools.   
For example this will colorize all `png` files in current directory and put them into `out` directory:
```sh
mkdir -p out; \
for f in *.png; do \
  convert "$f" -colorspace RGB -colorspace Gray +level-colors black,'#67d97a' "out/$f"; \
done
```

`#67d97a` is a font color used in `theme.txt`, you can replace it and corresponding HSL values with those you see fit, of course.

\* Easiest way to use BIMP on Linux is probably Flatpak, GIMP and BIMP are available on Flathub, see `flatpak search gimp`

**Comment by Teknoist** [comment]
![image](https://user-images.githubusercontent.com/37031361/204986251-b209a552-fc8d-4635-9432-b951fcc0acfc.png)
Thanks

**Comment by shvchk** [comment]
You are welcome : ) I've updated my comment above with more details.
Example:
```sh
convert k.png -colorspace RGB -colorspace Gray -brightness-contrast +65 +level-colors black,'#67d97a' k.colorized.png
```
![k colorized](https://user-images.githubusercontent.com/872945/205131141-1ea08c7d-d7b9-49ad-9c0c-b3e0e59b72d7.png)
